### PR TITLE
Validate types when creating `Key` object

### PR DESCRIPTION
**Pull request by cedrickcooke**

### CODE REVIEW

**Comment by cedrickcooke** [comment]
CC @davidtaylor-juul

**Review by davidtaylor-juul — APPROVED** [review]
LGTM

File: core/src/jsMain/kotlin/Key.kt [hunk 1]
```diff
@@ -2,6 +2,16 @@ package com.juul.indexeddb
 
 import com.juul.indexeddb.external.IDBKeyRange
 import kotlinext.js.jsObject
+import kotlin.js.Date
+
+private fun Array<dynamic>.validateKeyTypes() {
+    for (value in this) when (value) {
```
Comment (twyatt) [inline core/src/jsMain/kotlin/Key.kt:8]: TIL you can tack on `when` to the `for`. Pretty neat syntax.

**Review by twyatt — APPROVED** [review]

File: core/src/jsTest/kotlin/KeyTests.kt [hunk 2]
```diff
@@ -52,4 +52,9 @@ public class KeyTests {
     public fun constructor_withRange_completes() {
         Key(IDBKeyRange.upperBound("foobar", false))
     }
+
+    @Test
+    public fun constructor_withNull_completes() {
+        Key(null)
+    }
```
Comment (twyatt) [inline core/src/jsTest/kotlin/KeyTests.kt:59]: What is this test testing?

Comment (cedrickcooke) [reply core/src/jsTest/kotlin/KeyTests.kt:59]: This PR adds checks in the constructor for `Key` that the values passed in are legal types for IndexedDB. This test is just making sure that the check doesn't throw for `null`.

**Pull request merged**